sdk/nodejs/resource.ts

```diff
@@ -224,8 +224,11 @@ export abstract class Resource {
         // options assigned to this resource.
         const parent = opts.parent || getStackResource() || { __transformations: undefined };
         this.__transformations = [ ...(opts.transformations || []), ...(parent.__transformations || []) ];
+
+        let newname = name;
```

Comment:
Perhaps it is safer to use `name` as the updated name, so that code that refernces `name` picks up this new name by default.  I don't believe there is any code that should intentionally use the old name - but if there is - it seems we could use `const oldname = name` here to hold onto that before we modify `name`.  That also better matches what we do for `props` and `opts`. 

I'm afraid if we don't do that, we'll end up accidentally adding code here that references name and introduce subtle bugs.

Reply:
to make sure i'm understanding correctly, you are proposing we modify the constructor argument direct vs a mutable local variable?